plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -766,6 +779,57 @@ export class EventsProcessor {
         }
     }
 
+    private tryStoreSessionRecordingToObjectStorage(
+        timestamp: DateTime,
+        session_id: string,
+        snapshot_data: Record<any, any>,
+        team_id: number
+    ): Record<any, any> {
+        // As we don't want to store the session recording payload in ClickHouse,
+        // let's intercept the event, parse the metadata and store the data in
+        // our object storage system.
+
+        if (!this.objectStorage.isEnabled) {
+            return snapshot_data
+        }
+
+        const teamIsOnSessionRecordingToS3AllowList =
+            this.objectStorage.sessionRecordingAllowList === 'all' ||
+            this.objectStorage.sessionRecordingAllowList.includes(team_id)
+
+        if (!teamIsOnSessionRecordingToS3AllowList) {
+            return snapshot_data
+        }
+
+        const dateKey = castTimestampOrNow(timestamp, TimestampFormat.DateOnly)
+        const object_storage_path = `${OBJECT_STORAGE_SESSION_RECORDING_FOLDER}/${dateKey}/${session_id}/${snapshot_data.chunk_id}/${snapshot_data.chunk_index}`
+        const params = { Bucket: OBJECT_STORAGE_BUCKET, Key: object_storage_path, Body: snapshot_data.data }
+
+        const tags = {
+            team_id: team_id.toString(),
+            session_id,
+        }
+
+        const storageWriteTimer = new Date()
+
+        this.objectStorage.putObject(params, (err: any, resp: any) => {
```

Comment:
Plugin server works off of promises, not callbacks. Please create a promise-based API to be consistent.